app/src/main/java/org/ole/planet/myplanet/repository/RealmRepository.kt

```
@@ -131,6 +132,28 @@ open class RealmRepository(
+
+                val schema = newRealm.schema.get(clazz.simpleName)
+                val pkField = schema?.primaryKey
+                if (pkField == null) return@distinctUntilChanged false
+
+                for (i in 0 until old.size) {
+                    val oldObj = io.realm.DynamicRealmObject(old[i])
+                    val newObj = io.realm.DynamicRealmObject(new[i])
+                    if (oldObj.get<Any>(pkField) != newObj.get<Any>(pkField)) return@distinctUntilChanged false
+                }
+                true
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)